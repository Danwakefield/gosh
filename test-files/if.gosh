echo "6 test cases"
noop=true
if true; then
    echo "SUCCESS 1"
fi

A="FAIL 2"
if true; then
    if true; then
        A="SUCCESS 2"
    fi
fi
echo $A

A="SUCCESS 3"
if false; then
    A="FAIL 3"
fi
echo $A

A="FAIL 4"
if true
then
A="SUCCESS 4"
fi
echo $A

A="FAIL 5"
if false; then
    $noop
elif true; then
    A="SUCCESS 5"
fi
echo $A

A="FAIL 6"
if false; then
    $noop
elif false; then
    $noop
else
    A="SUCCESS 6"
fi
echo $A

