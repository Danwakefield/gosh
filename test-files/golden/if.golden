6 test cases
SUCCESS 1
SUCCESS 2
SUCCESS 3
SUCCESS 4
SUCCESS 5
SUCCESS 6
